src/PaymentGateways/Gateways/Stripe/StripePaymentElementGateway/Webhooks/Decorators/SubscriptionModelDecorator.php

```diff
@@ -26,65 +23,41 @@ public function __construct(Subscription $subscription)
     }
 
     /**
+     * @unreleased updated to use model method
      * @since 3.0.0
      */
     public function shouldEndSubscription(): bool
     {
-        return $this->subscription->installments !== 0 && (count(
-                    $this->subscription->donations
-                ) >= $this->subscription->installments);
+        return $this->subscription->shouldEndSubscription();
     }
 
     /**
+     * @unreleased updated to use model method
      * @since 3.0.0
      */
     public function shouldCreateRenewal(): bool
     {
-        $billTimes = $this->subscription->installments;
-        $totalPayments = count($this->subscription->donations);
-
-        return $this->subscription->status->isActive() && (0 === $billTimes || $totalPayments < $billTimes);
+        return $this->subscription->shouldCreateRenewal();
     }
 
     /**
+     * @unreleased updated to use model method
      * @since 3.0.0
      *
      * @throws Exception
      */
     public function handleRenewal(Invoice $invoice): SubscriptionModelDecorator
     {
-        $initialDonation = $this->subscription->initialDonation();
-
-        // create renewal
-        Donation::create([
+        $this->subscription->createRenewal([
             'amount' => Money::fromDecimal(
```

Comment:
@jonwaldstein It makes me wonder...  In Stripe, is it possible for a renewal to have a different value from the subscription? In which scenarios it can happen?

Reply:
@glaubersilva that's a great question!  Ideally there should not be a different value from the subscription in Give. My train of thought here was to use Stripe as the source of truth in case there is a different value - so we are recording the actual paid amount.  The 2 scenarios where this would be possible (yet, very unlikely) are:
1. The subscription was somehow updated in Stripe and not in Give
2. The amount was recorded wrong in Give

Reply:
@jonwaldstein Sounds good! As we discussed previously, we probably will need to refactor the `Give\Framework\PaymentGateways\Webhooks\EventHandlers\SubscriptionRenewalDonationCreated` class to support something like this. But we can do that in a subsequent PR.

Reply:
Sounds good, thanks!